packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2163,20 +2163,32 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
 
     final TextStyle hintStyle = _getInlineHintStyle(themeData, defaults);
     final String? hintText = decoration.hintText;
-    final Widget? hint = hintText == null ? null : AnimatedOpacity(
-      opacity: (isEmpty && !_hasInlineLabel) ? 1.0 : 0.0,
-      duration: decoration.hintFadeDuration ?? _kHintFadeTransitionDuration,
-      curve: _kTransitionCurve,
-      child: Text(
+    final bool hintIgnoreHeightOnInput = decoration.hintIgnoreHeightOnInput;
+    const Curve switchInCurve = Interval(0.4, 1.0, curve: Curves.fastOutSlowIn);
+    const Curve switchOutCurve = Interval(0.4, 1.0, curve: Curves.fastOutSlowIn);
+    Widget? hint;
+    if (hintText != null) {
+      final bool showHint = isEmpty && !_hasInlineLabel;
+      final Text hintTextWidget = Text(
         hintText,
         style: hintStyle,
         textDirection: decoration.hintTextDirection,
         overflow: hintStyle.overflow ?? (decoration.hintMaxLines == null ? null : TextOverflow.ellipsis),
         textAlign: textAlign,
         maxLines: decoration.hintMaxLines,
-      ),
-    );
-
+      );
+      hint = hintIgnoreHeightOnInput ? AnimatedSwitcher(
```

Comment:
I tried this out and unfortunately this doesn't work very well:
- There is no fade-out animation for the hint text.
- The size of the input decorator becomes too big when there's text.

Reply:
@LongCatIsLooong thx,  i update curve, and it is my test code.
```
 /// after
  buildTextField() {
    return Container(
      width: 200,
      decoration: BoxDecoration(
        border: Border.all(
          color: Colors.blue,
        ),
      ),
      child: TextField(
        controller: controller,
        minLines: 1,
        maxLines: 2,
        decoration: InputDecoration(
          hintText: 'To be, or not to be, that is question' * 10,
          hintIgnoreHeightOnInput: true,
          hintFadeDuration: const Duration(seconds: 1),
        ),
      ),
    );
  }
 /// beford
  buildTextField2() {
    return Container(
      width: 200,
      decoration: BoxDecoration(
        border: Border.all(
          color: Colors.green,
        ),
      ),
      child: TextField(
        controller: controller2,
        minLines: 1,
        maxLines: 2,
        decoration: InputDecoration(
          hintText: 'To be, or not to be, that is question' * 10,
          hintFadeDuration: const Duration(seconds: 1),
        ),
      ),
    );
  }
```

https://github.com/user-attachments/assets/08072d28-b29d-4ce4-aae6-06f0bb739dc9

Reply:
> I tried this out and unfortunately this doesn't work very well:
> 
> * There is no fade-out animation for the hint text.
> * The size of the input decorator becomes too big when there's input text, when hint text size is supposed to be (0, 0)

@LongCatIsLooong It seems that the hintFadeDuration time needs to be set, otherwise the fadeout effect will not be visible.

Reply:
Still it doesn't seem to fix the sizing. When there's no hint text the text field becomes even taller with this change.

Reply:
@LongCatIsLooong Sorry, I don't quite understand what you mean. Can you explain the problem in detail or provide the code I tested?

Reply:
Ah sorry I wasn't clear. This is my test app:
```dart
  runApp(MaterialApp(
    home: Scaffold(
      body: Center(
        child: TextField(
          decoration: InputDecoration(
            hintText: 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' * 5,
            hintMaxLines: 3,
            hintFadeDuration: Duration(seconds: 5),
          ),
        )
      ),
    ),
  ));
```
with this patch applied, the text field becomes much taller when I enter text into the text field, than it would be without the patch. Without the patch the text field would be roughly as tall as the characters. You can see that in widget inspectors or if you dump the render tree in the console.

Reply:
@LongCatIsLooong Thank you very much for the test code you provided, I fixed it.